src/devices/Common/Iot/Device/Common/WeatherHelper.cs

```diff
@@ -0,0 +1,210 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Iot.Units;
+using System;
+
+namespace Iot.Device.Common
+{
+    /// <summary>
+    /// Helpers for weather
+    /// </summary>
+    public class WeatherHelper
+    {
+        #region TemperatureAndRelativeHumidity
+        // Formulas taken from http://www.reahvac.com/tools/humidity-formulas/
+        // Extra notes for heat index taken from https://www.wpc.ncep.noaa.gov/html/heatindex_equation.shtml
+        
+        /// <summary>
+        /// The heat index (or apparent temperature) is used to measure the amount of discomfort
+        /// during the summer months when heat and humidity often combine to make it feel hotter
+        /// than it actually is. The heat index is usually used for afternoon high temperatures.
+        /// Use the summer simmer index instead for overnight low temperatures.
+        /// </summary>
+        /// <param name="airTemperature">The dry air temperature</param>
+        /// <param name="relativeHumidity">The relative humidity (RH) expressed as a percentage</param>
+        /// <returns>The heat index, also known as the apparent temperature</returns>
+        public static Temperature CalculateHeatIndex(Temperature airTemperature, double relativeHumidity)
+        {
+            double tf = airTemperature.Fahrenheit;
+            double rh = relativeHumidity;
+            double tf2 = Math.Pow(tf, 2);
+            double rh2 = Math.Pow(rh, 2);
+
+            var steadman = 0.5 * (tf + 61 + ((tf - 68) * 1.2) + (rh * 0.094));
+
+            if (steadman < 80)
+                return Temperature.FromFahrenheit(steadman);
+
+            var rothfuszRegression = (-42.379)
+                + (2.04901523 * tf)
```

Comment:
can this formula be simplified?

Reply:
Am following the actual linked formulas; it should be better to leave as is for clarity.

Reply:
sounds good